dev/com.ibm.ws.tx.embeddable/src/com/ibm/tx/jta/embeddable/impl/EmbeddableTransactionImpl.java

```diff
@@ -1178,6 +1191,7 @@ public String toString() {
         // NullPointerException when getId() is called
         Thread local_thread = _thread;
         return super.toString() + ",active=" + _activeAssociations + ",suspended=" + _suspendedAssociations + ","
-               + (local_thread != null ? "thread=" + String.format("%08X", local_thread.getId()) : "Not on a thread, globalId=" + _globalId);
+               + (local_thread != null ? "thread=" + String.format("%08X", local_thread.getId()) : "Not on a thread, globalId=" + _globalId + ", gtrid="
+                                                                                                   + Util.toHexString(_xid.getGlobalTransactionId()));
```

Comment:
I think the `_xid` may be null resulting in an NPE here.

I've only noticed this because one of our buckets has this JVM option and we are seeing a ton of java cores triggered by an NPE:

```
-Xdump:java:events=throw+systhrow,filter=java/lang/NullPointerException,request=exclusive+prepwalk+preempt
```

Where the current thread (that caused the NPE) is here:

```
1XMCURTHDINFO  Current thread
3XMTHREADINFO      "Default Executor-thread-3" J9VMThread:0x0000000000AC5A00, omrthread_t:0x00007F8390006E10, java/lang/Thread:0x00000006103C6500, state:R, prio=5
3XMJAVALTHREAD            (java/lang/Thread getId:0x41, isDaemon:true)
3XMJAVALTHRCCL            com/ibm/ws/classloading/internal/ThreadContextClassLoader(0x000000061055A140)
3XMTHREADINFO1            (native thread ID:0x1A885B, native priority:0x5, native policy:UNKNOWN, vmstate:R, vm thread flags:0x00041020)
3XMTHREADINFO2            (native stack address range from:0x00007F8426D80000, to:0x00007F8426E00000, size:0x80000)
3XMCPUTIME               CPU usage total: 0.811148751 secs, current category="Application"
3XMHEAPALLOC             Heap bytes allocated since last GC cycle=0 (0x0)
3XMTHREADINFO3           Java callstack:
4XESTACKTRACE                at com/ibm/tx/jta/embeddable/impl/EmbeddableTransactionImpl.toString(EmbeddableTransactionImpl.java:1195)
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter$3.run(BaseTraceFormatter.java:1151)
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter$3.run(BaseTraceFormatter.java:1148)
4XESTACKTRACE                at java/security/AccessController.doPrivileged(AccessController.java:692(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter.formatObject(BaseTraceFormatter.java:1148(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter.formatObj(BaseTraceFormatter.java:1087(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter.formatObj(BaseTraceFormatter.java:1072(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter.formatMessage(BaseTraceFormatter.java:311(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceFormatter.formatVerboseMessage(BaseTraceFormatter.java:381(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceService.publishTraceLogRecord0(BaseTraceService.java:1241)
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceService.publishTraceLogRecord(BaseTraceService.java:1221(Compiled Code))
4XESTACKTRACE                at com/ibm/ws/logging/internal/impl/BaseTraceService.entry(BaseTraceService.java:884)
4XESTACKTRACE                at com/ibm/websphere/ras/Tr.entry(Tr.java:589)
4XESTACKTRACE                at com/ibm/tx/jta/impl/LocalTIDTable.getLocalTID(LocalTIDTable.java:64)
```

Reply:
Interestingly enough, this doesn't seem to cause any errors in the FAT.  I assume the NPE must be caught by the tracing code somewhere and ultimately the `Tr.entry` must exist without actually doing anything.  I do have `Transaction=all` trace enabled in this test.

Reply:
Fixed by https://github.com/OpenLiberty/open-liberty/pull/28445